content/en-US/api/URLPattern.md

```diff
@@ -0,0 +1,515 @@
+const pattern = new URLPattern({
+  hostname: {*.}?example.com
+});
+
+// `{*.}?example.com`
+pattern.hostname;
+
+// `*`
+pattern.protocol;
+pattern.username;
+pattern.password;
+pattern.pathname;
+pattern.search;
+pattern.hash;
+
+// true
+pattern.test("https://example.com/foo/bar");
+
+// true
+pattern.test({ hostname: "cdn.example.com" });
+
+// true
+pattern.test("custom-protocol://example.com/other/path?q=1");
+
+// false because the hostname component does not match
+pattern.test("https://cdn-example.com/foo/bar");
+```
+
+### Example: Construct a URLPattern from a full URL string.
+
+The following example shows how a URLPattern can be constructed from a full
+URL string with patterns embedded.  If there is no ambiguity between whether
+a character is a URL separate or a pattern character then it just works.
+
+```js
+// Construct a URLPattern that matches URLs to CDN servers loading jpg images.
+// URL components not explicitly specified, like search and hash here, result
+// in the empty string similar to the URL() constructor.
+const pattern = new URLPattern("https://cdn-*.example.com/*.jpg");
+
+// `https`
+pattern.protocol;
+
+// `cdn-*.example.com`
+pattern.hostname;
+
+// `/*.jpg`
+pattern.pathname;
+
+// empty string
+pattern.username;
+pattern.password;
+pattern.search;
+pattern.hash;
+
+// true
+pattern.test("https://cdn-1234.example.com/product/assets/hero.jpg");
+
+// false because the search component does not match
+pattern.test("https://cdn-1234.example.com/product/assets/hero.jpg?q=1");
+```
+
+### Example: Constructing a URLPattern with an ambiguous URL string.
+
+The following example shows how a URLPattern constructed from an ambiguous
+string will favor treating characters as part of the pattern syntax.  In
+this case the `:` character could be the protocol component suffix or it
+could be the prefix for a pattern named group.  The constructor chooses
+to treat this as part of the pattern and therefore determines this is
+a relative pathname pattern.  Since there is no base URL the relative
+pathname cannot be resolved and it throws.
+
+```js
+// Throws because this is interpreted as a single relative pathname pattern
+// with a ":foo" named group and there is no base URL.
+const pattern = new URLPattern("data:foo*");
+```
+
+### Example: Escaping characters to disambiguate URLPattern constructor strings.
+
+The following example shows how an ambiguous constructor string character
+can be escaped to be treated as a URL separator instead of a pattern
+character.  Here `:` is escaped as `\\:`.
+
+```js
+// Constructs a URLPattern treating the `:` as the protocol suffix.
+const pattern = new URLPattern("data\\:foo*");
+
+// `data`
+pattern.protocol;
+
+// `foo*`
+pattern.pathname;
+
+// empty string
+pattern.username;
+pattern.password;
+pattern.hostname;
+pattern.port;
+pattern.search;
+pattern.hash;
+
+// true
+pattern.test("data:foobar");
+```
+
+### Example: Using base URLs for `test()` and `exec()`.
```

Comment:
```suggestion
### Using Base URLs for test() and exec()
```

Reply:
Special formatting for code keywords is often avoided in headings. In headings the disambiguation value of such formatting is minuscule and leaving it out makes a page visually more appealing.